### PR TITLE
hred: init at 1.5.0

### LINKED ISSUE #165720: Packaging request: hred
**Project description**
I've been searching for a tool to extract structured data usefully from web pages which don't provide a machine-readable interface such as RSS, and I recently found `hred`. `hred` extracts data from html using a language based on css selectors, but unlike other programs such as `pup` that do the same, it outputs its matches in (fairly configurable) json rather than as a flat list, and if that's not configurable enough, it can easily be piped into `jq` for more complex processing. It also apparently handles xml.

It seems better-thought-out than other tools in the category to me, and it doesn't seem to die horribly when exposed to real-world html either, like xml-oriented things often do.

I did try running `node2nix` in a local checkout, and the derivation it produced worked for me with no tweaking, so this is probably fairly trivial to package. I'm just not really familiar with how nixpkgs handles node stuff, so I'm not confident doing it myself.

**Metadata**

* homepage URL: https://github.com/danburzo/hred
* source URL: https://github.com/danburzo/hred
* license: mit
* platforms: anywhere node.js works

### PR DESCRIPTION
**Pull request by tejing1**
###### Description of changes

Packaged [hred](https://github.com/danburzo/hred), a command-line tool to extract data from HTML.

~~Given that this is a node package, of course the generate script has made changes affecting other packages, which I did not test.~~

Also added myself to the maintainers list. I went with a handle different from my github username, but if you really want me to change it, I will. Github is essentially the only place my name gets a `1` at the end, though. I'm `tejing` everywhere else.

Closes #165720

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by winterqt** [comment]
Can you package this with `buildNpmPackage` instead? See [here](https://nixos.org/manual/nixpkgs/unstable/#javascript-buildNpmPackage) for docs.

**Comment by tejing1** [comment]
That's much nicer for everyone. I'm not too familiar with how node stuff is done in nixpkgs, so I didn't know the lowest friction way to go about it.

**Comment by tejing1** [comment]
And I added a simple test.

**Review by winterqt — APPROVED** [review]
Thanks, and welcome! :)

**Comment by winterqt** [comment]
Oh, one thing: can you add an update script? This should suffice:

```nix
passthru.updateScript = nix-update-script { };
```

**Comment by SuperSandro2000** [comment]
Adding an update script is not necessary for r-ryantm to work if standard fetchers are used.

**Comment by winterqt** [comment]
> Adding an update script is not necessary for r-ryantm to work if standard fetchers are used.

In this case, it is -- r-ryantm does not support buildNpmPackage as of writing.

**Comment by mweinelt** [comment]
> Adding an update script is not necessary for r-ryantm to work if standard fetchers are used.

You are clueless and ignoring requests by other commiters. Stop that behaviour immediately.

File: pkgs/development/tools/hred/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  version = "1.5.0";
+
+  src = fetchFromGitHub {
+    owner = "danburzo";
+    repo = "hred";
+    rev = "v${version}";
+    hash = "sha256-rnobJG9Z1lXEeFm+c0f9OsbiTzxeP3+zut5LYpGzWfc=";
+  };
+
+  npmDepsHash = "sha256-POxlGWK0TJMwNWDpiK5+OXLGtAx4lFJO3imoe/h+7Sc=";
```
Comment (mweinelt) [inline pkgs/development/tools/hred/default.nix:14]: This is what r-ryantm does not support yet.

**Comment by tejing1** [comment]
Eh, I'll live without the bot's help on the next version upgrade :-)

I'll try to remember to add `passthru.updateScript` when that comes, though.

**Comment by SuperSandro2000** [comment]
> In this case, it is -- r-ryantm does not support buildNpmPackage as of writing.

Thats good to know.

> You are clueless and ignoring requests by other commiters.

It would help me a lot if those are phrased in such a way that it is obvious that they are blocking and not just nice to have additions.

**Comment by winterqt** [comment]
> It would help me a lot if those are phrased in such a way that it is obvious that they are blocking and not just nice to have additions.

You could alternatively just let me handle the PR that I've reviewed... it's not like it's been a week without me not responding (which I'm not saying I haven't done) or anything. Or ask me to clarify, and don't assume that I'm wrong.

**Pull request merged**